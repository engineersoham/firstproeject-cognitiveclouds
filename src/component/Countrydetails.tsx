import React, { useState } from 'react'
import { Box, Paper } from '@mui/material';
import { useParams } from 'react-router-dom';
import { useEffect } from 'react';
import { TableRow, TableCell, IconButton, Stack, Avatar, Button } from '@mui/material';
import { Link } from 'react-router-dom';


const Countrydetails = () => {
    const { country } = useParams()
    const [contryDetailes, setDetailes] = useState({
        countryname: '',
        Capital: '',
        population: 0,
        latitude: 0,
        longitude: 0,
        Flag: '',
        imageurl: ''
    })

    const detailes = async () => {
        const result = await fetch(`https://restcountries.com/v3.1/name/${country}`);
        const [data] = await result.json();
        setDetailes({
            ...contryDetailes,
            countryname: data.name.common,
            Capital: data.capital[0],
            population: data.population,
            latitude: data.capitalInfo.latlng[0],
            longitude: data.capitalInfo.latlng[1],
            Flag: data.flags.png,
            imageurl: data.coatOfArms.png
        })
        console.log(data)
    }

    useEffect(() => {
        try{
            detailes()
        }
        catch(error){
            alert('Enter correct country name')
        }
        
    }, [country])
    return (
        <div  >

            <h3 style={{ textAlign: 'center' }}>Country Details</h3>
            <Paper style={{ boxShadow: '5px 10px #888888', border: '1px solid black', width: '25rem', height: '13rem' }} sx={{ p: 5, ml: 65, mr: 70, mt: 10 }}>
                <div style={{ float: 'left' }}>
                    <table  style={{ marginBottom: '3rem' }}>
                        <tr>
                            <td style={{ textAlign: 'left' }}> CountryName  </td>
                            <td style={{ textAlign: 'left' }}>: <b>{contryDetailes.countryname}</b></td>

                        </tr>
                        <tr>
                            <td style={{ textAlign: 'left' }}>Capital </td>
                            <td style={{ textAlign: 'left' }}>: <b>{contryDetailes.Capital}</b></td>
                        </tr>
                        <tr>
                            <td style={{ textAlign: 'left' }}>Population </td>
                            <td style={{ textAlign: 'left' }}>: <b>{contryDetailes.population}</b></td>
                        </tr>
                        <tr>
                            <td style={{ textAlign: 'left' }}>Latitude </td>
                            <td style={{ textAlign: 'left' }}>: <b>{contryDetailes.latitude}</b></td>

                        </tr>
                        <tr>
                            <td style={{ textAlign: 'left' }}>Longitude </td>
                            <td style={{ textAlign: 'left' }}>: <b>{contryDetailes.longitude}</b></td>

                        </tr>
                    </table>
                    <Link to={`/weather/${contryDetailes.Capital}`} style={{ textDecoration: 'none' }}>
                        <IconButton aria-label="delete" size="large" >
                            <Button variant="contained">Capital Weather</Button>
                        </IconButton>
                    </Link>
                </div>

                <div style={{ float: 'left' }}>
                    <Avatar
                        variant="rounded"
                        src={contryDetailes.Flag}
                        sx={{ width: 70, height: 70, mt: 7, ml: 10 }}
                    />

                </div>


            </Paper>


            {/* <Box 
                sx={{ml:10,mt:10,
                    display: 'non',
                    flexWrap: 'wrap',
                    '& > :not(style)': {
                        m: 2,
                        width: 1300,
                        height: 100,
                    },
                }}
            >
                <Paper elevation={5}  style={{backgroundImage: 'linear-gradient(to left, rgba(210,0,0,0), rgba(230,0,0,0.5))'}} >
                    <TableRow hover sx={{ mb: 3, '&:last-child td, &:last-child th': { border: 0 } }} >
                        <TableCell component="th" scope="row">
                            <Button variant="outlined">
                                {contryDetailes.countryname}.
                            </Button>
                        </TableCell>
                        <TableCell>
                            <Stack direction="row" alignItems="center" spacing={1}>
                                <Button variant='outlined' color="secondary">{contryDetailes.Capital}</Button>
                            </Stack>
                        </TableCell>
                        <TableCell>
                        </TableCell>
                        <TableCell>
                            <IconButton aria-label="delete" size="large" >
                                <tr sx={{ mr: 2 }}>
                                    <Button color="error">
                                        population: {contryDetailes.population}
                                    </Button>
                                </tr>
                            </IconButton>
                            <IconButton aria-label="delete" size="large" >
                                <tr>
                                    Latitude: {contryDetailes.latitude}
                                </tr>
                                <tr sx={{ mr: 5, ml: 2 }}>
                                    Longitude: {contryDetailes.longitude}
                                </tr>
                            </IconButton>
                            <IconButton aria-label="delete" size="large" >
                                <Avatar
                                    variant="rounded"
                                    src={contryDetailes.Flag}
                                    sx={{ width: 56, height: 56 }}
                                />
                            </IconButton>
                            <Link to={`/weather/${contryDetailes.Capital}`} style={{ textDecoration: 'none' }}>
                                <IconButton sx={{ ml: 10 }} aria-label="delete" size="large" >
                                    <Button variant="contained">Capital Weather</Button>
                                </IconButton>
                            </Link>
                        </TableCell>
                    </TableRow>
                   
                </Paper>
            </Box> */}


        </div>
    )
}

export default Countrydetails