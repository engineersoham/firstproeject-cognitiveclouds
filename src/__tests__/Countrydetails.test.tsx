import {render, screen} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Countrydetails from '../component/Countrydetails';

it('country component test', () => {
    const country = render(
        <BrowserRouter>
        <Countrydetails/>
        </BrowserRouter>
    );
    expect(country).toMatchSnapshot()
})

describe('test for country details', () => {
    it('rendering detail country component', () => {
      render(
        <BrowserRouter>
           <Countrydetails/>
        </BrowserRouter>
      );
      screen.debug();
    })
  })
 