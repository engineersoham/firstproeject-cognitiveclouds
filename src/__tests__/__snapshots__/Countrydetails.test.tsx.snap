// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`country component test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h3
          style="text-align: center;"
        >
          Country Details
        </h3>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-7vqw9k-MuiPaper-root"
          style="box-shadow: 5px 10px #888888; border: 1px solid black; width: 25rem; height: 13rem;"
        >
          <div
            style="float: left;"
          >
            <table
              style="margin-bottom: 3rem;"
            >
              <tr>
                <td
                  style="text-align: left;"
                >
                   CountryName  
                </td>
                <td
                  style="text-align: left;"
                >
                  : 
                  <b />
                </td>
              </tr>
              <tr>
                <td
                  style="text-align: left;"
                >
                  Capital 
                </td>
                <td
                  style="text-align: left;"
                >
                  : 
                  <b />
                </td>
              </tr>
              <tr>
                <td
                  style="text-align: left;"
                >
                  Population 
                </td>
                <td
                  style="text-align: left;"
                >
                  : 
                  <b>
                    0
                  </b>
                </td>
              </tr>
              <tr>
                <td
                  style="text-align: left;"
                >
                  Latitude 
                </td>
                <td
                  style="text-align: left;"
                >
                  : 
                  <b>
                    0
                  </b>
                </td>
              </tr>
              <tr>
                <td
                  style="text-align: left;"
                >
                  Longitude 
                </td>
                <td
                  style="text-align: left;"
                >
                  : 
                  <b>
                    0
                  </b>
                </td>
              </tr>
            </table>
            <a
              href="/weather/"
              style="text-decoration: none;"
            >
              <button
                aria-label="delete"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Capital Weather
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
          <div
            style="float: left;"
          >
            <div
              class="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault css-3q23wa-MuiAvatar-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
                data-testid="PersonIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h3
        style="text-align: center;"
      >
        Country Details
      </h3>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-7vqw9k-MuiPaper-root"
        style="box-shadow: 5px 10px #888888; border: 1px solid black; width: 25rem; height: 13rem;"
      >
        <div
          style="float: left;"
        >
          <table
            style="margin-bottom: 3rem;"
          >
            <tr>
              <td
                style="text-align: left;"
              >
                 CountryName  
              </td>
              <td
                style="text-align: left;"
              >
                : 
                <b />
              </td>
            </tr>
            <tr>
              <td
                style="text-align: left;"
              >
                Capital 
              </td>
              <td
                style="text-align: left;"
              >
                : 
                <b />
              </td>
            </tr>
            <tr>
              <td
                style="text-align: left;"
              >
                Population 
              </td>
              <td
                style="text-align: left;"
              >
                : 
                <b>
                  0
                </b>
              </td>
            </tr>
            <tr>
              <td
                style="text-align: left;"
              >
                Latitude 
              </td>
              <td
                style="text-align: left;"
              >
                : 
                <b>
                  0
                </b>
              </td>
            </tr>
            <tr>
              <td
                style="text-align: left;"
              >
                Longitude 
              </td>
              <td
                style="text-align: left;"
              >
                : 
                <b>
                  0
                </b>
              </td>
            </tr>
          </table>
          <a
            href="/weather/"
            style="text-decoration: none;"
          >
            <button
              aria-label="delete"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Capital Weather
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
        <div
          style="float: left;"
        >
          <div
            class="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault css-3q23wa-MuiAvatar-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
              data-testid="PersonIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;